### ISSUE TITLE
Automate deploy with Docker

### ISSUE THREAD

**Issue by cuducos** [datasciencebr/jarbas#53]
Now we have a Docker structure, so we can think about how to use that to automate the deploy.

Ideally a piece of code to automate provision is much appreciated ; )

**Comment by vmesel** [comment]
Docker is easy, I can handle this one!

**Comment by cuducos** [comment]
@gomex might be workin on or planning something in that direction — you both could collaborate with each other if that's the case ; )

**Comment by gwmoura** [comment]
@vmesel You have one idea, how to deploy? Today we need to deploy 4 services (maybe 3):

* jarbas - django app
* elm - nodejs
* web - nginx
* db - postgresql

**Comment by pedrommone** [comment]
I think a `Ansible` is the easiest way to provision it right now, since the stack is very tiny. How are you planning it @gomex?

**Comment by gomex** [comment]
I was thinking about using ansible to create the docker host and to run the docker-compose there. What you think about?

**Comment by pedrommone** [comment]
@gomex it's the easiest way and probably the best way for now.

**Comment by gwmoura** [comment]
Hi, guys how this is running?

**Comment by cuducos** [comment]
@gwmoura Apparently nobody's working on that — ages ago I started a branch called [`ansible`](https://github.com/datasciencebr/jarbas/tree/ansible) but back then it was focused on deploying without Docker. I'd be happy to pair on that to sort that out, but I need guidance from one of you guys.

**Comment by gwmoura** [comment]
@cuducos we can pair, but basically is use the ansible to install docker on server and run `docker-compose up`.

We can follow the steps on this link: https://docs.docker.com/compose/production/

* Create a `jarbas-production.yaml`
* Define tasks to be runned before app run
* Define strategy for no downtime :smile:  
   > to deploy a container is necessary stop one before start another - :sweat_smile:

**Comment by cuducos** [comment]
Great material. I have the feeling that I will take a lot of time with that with the less basic stuff (such as keep sensible information on the environment etc). It's not my priority right now but I'd make it a priority if I can pair. Otherwise I'll try to build a `Dockerfile` for [Rosie](https://github.com/datasciencebr/rosie), so Rosie can live in the same server as Jarbas, easily deployed with Docker ; )

**Comment by gomex** [comment]
I can pair with you. I already have a ansible role to create a droplet on Digital Ocean with Docker host installed and now we just need to configure a role to start docker there.

**Comment by cuducos** [comment]
Great! When does it work to you? I'm basically free weekdays, whole day, except 2-4pm ; )

**Comment by gomex** [comment]
I am free weekdays too, but in the night is better for me :)

**Comment by gomex** [comment]
I am finish to clean the code and send I PR with a proposal for create droplets on Digital Ocean using ansible.

**Comment by gomex** [comment]
I am doing that in my fork, you can follow there:

https://travis-ci.org/gomex/jarbas

My problem now is about keep together the old deploy code of travis and the new proposal to use ansible, because ansible is not ok yet to use 3.5 python language :( I changed to 2.7 and got problem with that:

```
coverage run manage.py test -v 3
Couldn't read config file .coveragerc: File contains no section headers.
file: .coveragerc, line: 1
'  [run]\n'
The command "coverage run manage.py test -v 3" exited with 1.
```

**Comment by cuducos** [comment]
Unfortunately we need Python 3 to run Jarbas. If you start travis in Python 2.7, you can install (if needed, have to check Ubuntu/Debian version) `python3` and run all the commands with `python3` instead of `python` (for example `python3 manage.py test` instead of `python manage.py test`) — sounds reasonable?

**Comment by gomex** [comment]
@cuducos today the jarbas is deployed in only one machine or a cluster?

**Comment by cuducos** [comment]
Only one DO droplet ; )

**Comment by lipemorais** [comment]
I believe that the next step after the PR about make Docker work for Jarbas development(a.k.a #163 ) could this one. What do you think @cuducos ?

**Comment by cuducos** [comment]
So far so good @lipemorais — I'm gonna test and (hopefully) merge #163 today and then we can look into deploy with Docker <3

**Comment by cuducos** [comment]
Closed in favor of #191
